superset-frontend/src/components/Chart/DrillBy/DrillByModal.tsx

```diff
@@ -175,7 +209,48 @@ export default function DrillByModal({
         `}
       >
         {metadataBar}
-        <DrillByChart formData={updatedFormData} />
+        <div
+          css={css`
+            margin-bottom: ${theme.gridUnit * 6}px;
+            .ant-radio-button-wrapper-checked:not(.ant-radio-button-wrapper-disabled):focus-within {
+              box-shadow: none;
+            }
+          `}
+        >
+          <Radio.Group
+            onChange={({ target: { value } }) => {
+              setShowChart(value === DrillByType.chart);
+            }}
+            defaultValue={DrillByType.chart}
+          >
+            <Radio.Button
+              value={DrillByType.chart}
+              data-test="drill-by-chart-radio"
+            >
+              {t('Chart')}
+            </Radio.Button>
+            <Radio.Button
+              value={DrillByType.table}
+              data-test="drill-by-table-radio"
+            >
+              {t('Table')}
+            </Radio.Button>
+          </Radio.Group>
+        </div>
+        {showChart && chartDataResult && (
+          <DrillByChart formData={updatedFormData} result={chartDataResult} />
+        )}
+        {!showChart && chartDataResult && (
+          <SingleQueryResultPane
+            data={chartDataResult[0].data}
+            colnames={chartDataResult[0].colnames}
+            coltypes={chartDataResult[0].coltypes}
+            datasourceId={datasourceId}
+            dataSize={DATA_SIZE}
+            key={1}
+            isVisible={!showChart}
```

Comment:
nit: `isVisible` will always be true (because otherwise we won't render the component - line 243) so maybe we could just use `isVisible` without the condition here?